### PR TITLE
kalphitequeen: plugin

### PR DESCRIPTION
**Pull request by dutta64**

### CODE REVIEW

**Comment by CJP10** [comment]
LGTM, although you might  want to add a config option so that people can add their own animations to the animation map.
Something like menu entry swapper textbox and maybe a syntax like `<anim id>:<ticks>`

**Comment by dutta64** [comment]
Good call, I’ll add that.

**Comment by CJP10** [comment]
Another note, I think the player attack timing is useful as a standalone plugin. If that makes sense to you, let's just add that onto this PR. I can help with the data mining of animations for common weapons.

**Comment by CJP10** [comment]
https://github.com/xKylee/plugins-source/pull/221

File: kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+		for (final NPC npc : client.getNpcs())
+		{
+			addNpc(npc);
+		}
+
+		eventBus.subscribe(GameTick.class, this, this::onGameTick);
+		eventBus.subscribe(NpcSpawned.class, this, this::onNpcSpawned);
+		eventBus.subscribe(NpcDespawned.class, this, this::onNpcDespawned);
+		eventBus.subscribe(VarbitChanged.class, this, this::onVarbitChanged);
```
Comment (xKylee) [inline kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java:234]: I dont believe this is needed.

File: kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+			return;
+		}
+
+		init();
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		eventBus.unregister(this);
```
Comment (xKylee) [inline kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java:110]: I dont believe this is needed.

File: kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+				break;
+			case LOGIN_SCREEN:
+			case HOPPING:
+				if (atKalphiteQueen)
+				{
+					shutDown();
+				}
+				break;
+		}
+	}
```
Comment (xKylee) [inline kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java:154]: this is kinda redundant

File: kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+	private void onGameTick(final GameTick event)
+	{
```
Comment (xKylee) [inline kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java:157]: do a region check here.

File: kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+
+		eventBus.subscribe(GameTick.class, this, this::onGameTick);
+		eventBus.subscribe(NpcSpawned.class, this, this::onNpcSpawned);
+		eventBus.subscribe(NpcDespawned.class, this, this::onNpcDespawned);
+		eventBus.subscribe(VarbitChanged.class, this, this::onVarbitChanged);
+	}
+
+	private boolean atKalphiteQueen()
+	{
+		return client.getMapRegions()[0] == KQ_REGION;
```
Comment (xKylee) [inline kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQPlugin.java:239]: this doesn't account for multiple scene planes. it should only account for the scene you are on.

File: kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQSceneOverlay.java [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+	@Inject
+	KQSceneOverlay(final KQPlugin plugin, final KQConfig config, final Client client)
+	{
+		this.plugin = plugin;
+		this.config = config;
+		this.client = client;
+
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.HIGH);
+		setLayer(OverlayLayer.UNDER_WIDGETS);
```
Comment (xKylee) [inline kalphitequeen/src/main/java/net/runelite/client/plugins/kalphitequeen/KQSceneOverlay.java:73]: determineLayer();

add mirror mode support.

**Comment by dutta64** [comment]
@xKylee changes made

**Pull request merged**